packages/extension-utils/src/OAuthForm/src/state/sagas.ts

```diff
@@ -0,0 +1,201 @@
+        newMessage: error,
+      })
+    )
+  } finally {
+    // update the form element
+    const url = validateUrl(action.payload.value)
+    yield* put(hanldeUrlChangeSuccess(url || action.payload.value))
+  }
+}
+
+/**
+ * clears form, removes local storage config key and triggers callback function
+ */
+function* clearFormSaga(action: PayloadAction<ClearFormActionPayload>) {
+  const { clearFormActionSuccess, setFailureAction, updateMessageBarAction } =
+    OAuthFormActions
+  try {
+    const { configKey, setHasConfig, isAuthenticated } = action.payload
+    localStorage.removeItem(configKey)
+    if (setHasConfig) setHasConfig(false)
+    if (isAuthenticated) {
+      yield* put(
+        updateMessageBarAction({
+          intent: 'warn',
+          text: 'Please reload the browser page to log out',
+        })
+      )
+    }
+    yield* put(clearFormActionSuccess())
+  } catch (error: any) {
+    yield* put(setFailureAction(error.message))
+  }
+}
+
+/**
+ * verify button clicked, verifys apiServerUrl and populates OAuth server URL if valid
+ */
+function* handleVerifySaga() {
+  const apiServerUrl = yield* select(
+    (state: OAuthFormState) => state.apiServerUrl
+  )
+  const {
+    handleVerifyActionFailure,
+    handleVerifyActionSuccess,
+    clearMessageBarAction,
+  } = OAuthFormActions
+
+  try {
+    yield* put(clearMessageBarAction())
+    const versionsUrl = `${apiServerUrl}/versions`
+
+    const versions = yield* call(getVersions, versionsUrl)
+    if (!versions) throw new Error()
+
+    yield* put(handleVerifyActionSuccess(versions.web_server_url))
+  } catch (error: any) {
+    yield* put(handleVerifyActionFailure(error.message))
+  }
+}
+
+/**
+ * save button clicked, verify api server url and if valid save config data to localstorage
+ */
+function* handleSaveConfigSaga(action: PayloadAction<SaveConfigPayload>) {
+  const apiServerUrl = yield* select(
+    (state: OAuthFormState) => state.apiServerUrl
+  )
+  const {
+    handleVerifyActionFailure,
+    clearMessageBarAction,
+    handleSaveConfigSuccess,
+  } = OAuthFormActions
+  const { configKey, setHasConfig, client_id, redirect_uri } = action.payload
+
+  try {
+    yield* put(clearMessageBarAction())
+    const versionsUrl = `${apiServerUrl}/versions`
+
+    const versions = yield* call(getVersions, versionsUrl)
+    if (!versions) throw new Error()
+
+    const { api_server_url, web_server_url } = versions
+    const data = {
+      base_url: api_server_url,
+      looker_url: web_server_url,
+      client_id,
+      redirect_uri,
+    }
+    localStorage.setItem(configKey, JSON.stringify(data))
+    if (setHasConfig) setHasConfig(true)
+
+    yield* put(
+      handleSaveConfigSuccess({
+        base_url: api_server_url,
+        looker_url: web_server_url,
+      })
+    )
+  } catch (error: any) {
+    yield* put(handleVerifyActionFailure(error.message))
+  }
+}
+
+export function* saga() {
+  const {
+    initAction,
+    handleUrlChangeAction,
```

Comment:
Should I drop the 'handle' from these actions?

Reply:
you could change them to `setUrlAction`, `verifyAction`, `saveConfigAction`, `clearConfigAction` respectively. The handlers themselves should follow the naming convention `handle*`.